command-parser.js

```diff
@@ -209,9 +204,11 @@ var parse = exports.parse = function (message, room, user, connection, levelsDee
 	var commandHandler;
 	do {
 		commandHandler = currentCommands[cmd];
-		if (typeof commandHandler === 'string') {
-			// in case someone messed up, don't loop
+		var failsafe = 0;
+		while (typeof commandHandler === 'string') {
+			if (failsafe >= MAX_PARSE_RECURSION) return connection.sendTo(room, "Error: Too much recursion");
```

Comment:
Like I said elsewhere, I'd recommend recursion in function calls rather than recursive aliases.